primitives/arithmetic/src/per_things.rs

```diff
@@ -496,11 +498,29 @@ where
 	(x / maximum) * part_n + c
 }
 
+/// Wrapper of checked function
+fn rational_mul_correction<N, P>(x: N, numer: P::Inner, denom: P::Inner, rounding: Rounding) -> N
+where
+	N: MultiplyArg + UniqueSaturatedInto<P::Inner>,
+	P: PerThing,
+	P::Inner: Into<N>,
+{
+	checked_rational_mul_correction::<N, P>(x, numer, denom, rounding)
+		.expect("tries to divide by zero")
+}
+
 /// Compute the error due to integer division in the expression `x / denom * numer`.
 ///
 /// Take the remainder of `x / denom` and multiply by  `numer / denom`. The result can be added
 /// to `x / denom * numer` for an accurate result.
-fn rational_mul_correction<N, P>(x: N, numer: P::Inner, denom: P::Inner, rounding: Rounding) -> N
+///
+/// Checking if &denon is zero should cover denom_n and denom_upper
+fn checked_rational_mul_correction<N, P>(
```

Comment:
I am not sure here either, but as you said in chat; we could also make all function checked and then rename the ones that can panic to `unchecked_`.  
Then developers would use the safe variants per default.

Reply:
yeah, as I didn't have confirmation, I wasn't sure if that was the best approach, but I can flip it easily.